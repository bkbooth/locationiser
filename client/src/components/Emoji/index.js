export { default, emojis } from './Emoji';
