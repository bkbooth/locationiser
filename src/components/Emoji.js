import React from 'react';

export const emojis = {
  wave: { emoji: 'ðŸ‘‹', label: 'waving hand' },
};

function Emoji({ emoji: { emoji, label } }) {
  return (
    <span role="img" aria-label={label}>
      {emoji}
    </span>
  );
}

export default Emoji;
