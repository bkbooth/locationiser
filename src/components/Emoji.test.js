import React from 'react';
import { render } from '@testing-library/react';
import Emoji, { emojis } from './Emoji';

it('renders a pin emoji', () => {
  const { container } = render(<Emoji emoji={emojis.pin} />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});

it('renders a waiting emoji', () => {
  const { container } = render(<Emoji emoji={emojis.waiting} />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});

it('renders a waving emoji', () => {
  const { container } = render(<Emoji emoji={emojis.wave} />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});

it('renders a world map emoji', () => {
  const { container } = render(<Emoji emoji={emojis.worldMap} />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});
