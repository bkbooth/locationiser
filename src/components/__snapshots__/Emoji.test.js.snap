// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a pin emoji 1`] = `
<div>
  <span
    aria-label="round pushpin"
    role="img"
  >
    📍
  </span>
</div>
`;

exports[`renders a waiting emoji 1`] = `
<div>
  <span
    aria-label="hourglass not done"
    role="img"
  >
    ⏳
  </span>
</div>
`;

exports[`renders a waving emoji 1`] = `
<div>
  <span
    aria-label="waving hand"
    role="img"
  >
    👋
  </span>
</div>
`;

exports[`renders a world map emoji 1`] = `
<div>
  <span
    aria-label="world map"
    role="img"
  >
    🗺️
  </span>
</div>
`;
